src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -594,11 +603,119 @@ def get_household_structure(pop: pd.DataFrame) -> pd.DataFrame:
         return household_structure
 
     @staticmethod
-    def choose_random_guardian(member_ids: pd.DataFrame) -> pd.Series:
+    def choose_random_guardian(member_ids: pd.DataFrame, groupby_level: str) -> pd.Series:
         # member_ids is a subset of child_households dataframe
+        # groupby_level will be index level to group by (the first level index of member_ids (child_id or mother_id).
         member_ids = (
             member_ids.reset_index()
-            .groupby(["child_id"])["person_id"]
+            .groupby([groupby_level])["person_id"]
             .apply(np.random.choice)
         )
         return member_ids
+
+    def assign_second_guardian_to_newborns(
+        self, new_births: pd.DataFrame, households: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        new_births: pd.DataFrame of new births on this time step
+        households: pd.DataFrame of the entire state table for the two columns of household_id
+          and relation_to_household_head.
+
+        Returns
+        -------
+        new_births with the additional guardian_2 column updated.
+        """
+        # Setup
+        new_births["guardian_2"] = data_values.UNKNOWN_GUARDIAN_IDX
+        mothers_households = self.get_mothers_household_structure(
+            new_births["parent_id"], households
+        )
+        partners = [
```

Comment:
I'd suggest defining all these relationship lists as constants outside the functions, since you're using the same list of relationships for partners in both functions now.